search.py

```diff
@@ -336,42 +340,44 @@ def extend(U, open_dir, open_other, g_dir, g_other, closed_dir):
         open_dir.remove(n)
         closed_dir.append(n)
 
-        for c in problem.actions(n):
-            if c in open_dir or c in closed_dir:
-                if g_dir[c] <= problem.path_cost(g_dir[n], n, None, c):
+        for child in problem.actions(n):
```

Comment:
I am very confused here! Why is the child present in `problem.actions(n)` instead of `n.expand(problem)`? 
Why does `problem.action(n)` return the next states instead of actions possible in that states?

Reply:
Inspecting further, it turns out that bidirectional search only supports `GraphProblem` type problems. I think we can add support for all the problem types by storing the nodes instead of states in open/closed list for both direction and calling n.expand(problem) on it to return the next states. Any other ideas?

Reply:
`problem.actions(n)` does return the actions possible in `n`, but since this is a graph problem, the actions possible are the next states. The function simply returns the neighbouring nodes, which are both the next states and actions possible. I believe though that you are right, if we want to support other types of problems, we need to switch to `n.expand(problem)` which is more general.

If you want, you can take this up, making bidirectional search general. That sounds like a great contribution!

Reply:
Yup, will fix that!